### DISCUSSION TITLE
Why do I only get a parent through setTimeout?

### DISCUSSION THREAD

**Discussion by jic8** [pixijs/pixijs#9372] [Q&A] [answered]
`export default class CardPriceOther extends Container {
  constructor() {
    super();
    console.log(this);
    console.log(this.parent);
  }
}
`
In the console I get:
<details><summary>console.log(this)</summary>
<p>
Card {_events: Events, _eventsCount: 0, tempDisplayObjectParent: null, transform: _Transform, alpha: 1, …}
alpha: 1
children: []
parent: _Container {_events: Events, _eventsCount: 0, tempDisplayObjectParent: null, transform: _Transform, alpha: 1, …}
... more ... 
</p>
</details> 
 console.log(this.parent);  // null
 
 I have no problem getting, for example, this.heights, but the only way I can get parent this is `setTimeout(() => console.log(this.parent),0);`

**Comment by SuperSodaSea** [comment]
Obviously the `parent` in constructor is always null, since the `addChild` occurs after the constructor is completed. The problem is that the value of the object field in the console is not necessarily the value at the time `console.log` is executed. The console may evaluate its value later when you open the console after `console.log`, or when expanding the details of the object.

For example, save the following code to a .html file and open it, then open the console and see what is the output. You may see different results when you refresh the page while the console is open.

```html
<!DOCTYPE html>
<html>
    <head>
        <title>Test</title>
    </head>
    <body>
        <script>
            class Foo {
                constructor() {
                    this.a = 1;
                    console.log(this);
                    console.log(this.a);
                    this.a = 2;
                }
            }
            new Foo();
        </script>
    </body>
</html>
```

**Reply by jic8** [reply]
Thanks for the reply. I realised that, and that's why I used timeout. How this mechanic is more of a bug than a feature. Are there any recipes for getting the parent in the class constructor in pixi?  Or do I need to write my own bicycle?

**Reply by SuperSodaSea** [reply]
Unless you pass parent as an argument to the constructor, you can't get the object's parent in the constructor.

```js
const obj = new CustomObject();
container.addChild(obj); // We can only know what is obj's parent after addChild

const obj = new CustomObject(container); // This is OK
```

**Reply by bigtimebuddy** [reply, marked as answer]
You can add an event handler in the constructor. There's an event called `added` when a DisplayObject is added to a container. I'd use this if you need a reference to the parent.

**Reply by jic8** [reply]
> You can add an event handler in the constructor. There's an event called `added` when a DisplayObject is added to a container. I'd use this if you need a reference to the parent.

Thank you for your reply. I don't know how I didn't see that in the documentation.  I just replaced setTimeout and everything worked as it should.
